src/SAML2/XML/xenc/EncryptedData.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\xenc;
+
+use DOMElement;
+use SAML2\XML\ds\KeyInfo;
+use Webmozart\Assert\Assert;
+
+/**
+ * Class containing encrypted data.
+ *
+ * Note: <xenc:EncryptionProperties> elements are not supported.
+ *
+ * @package simplesamlphp/saml2
+ */
+class EncryptedData extends AbstractXencElement
+{
+    /** @var CipherData */
+    protected $cipherData;
+
+    /** @var string|null */
+    protected $encoding;
+
+    /** @var EncryptionMethod|null */
+    protected $encryptionMethod;
+
+    /** @var string|null */
+    protected $id;
+
+    /** @var KeyInfo|null */
+    protected $keyInfo;
+
+    /** @var string|null */
+    protected $mimeType;
+
+    /** @var string|null */
+    protected $type;
+
+
+    /**
+     * EncryptedData constructor.
+     *
+     * @param CipherData $cipherData The CipherData object of this EncryptedData.
+     * @param string|null $id The Id attribute of this object. Optional.
+     * @param string|null $type The Type attribute of this object. Optional.
+     * @param string|null $mimeType The MimeType attribute of this object. Optional.
+     * @param string|null $encoding The Encoding attribute of this object. Optional.
+     * @param EncryptionMethod|null $encryptionMethod The EncryptionMethod object of this EncryptedData. Optional.
+     * @param KeyInfo|null $keyInfo The KeyInfo object of this EncryptedData. Optional.
+     */
+    public function __construct(
+        CipherData $cipherData,
+        ?string $id,
```

Comment:
Shouldn't all of these nullables be optional?

Reply:
Good catch!

Reply:
Not just here, also in other classes

Reply:
As far as I've seen, `EncryptedKey` as well. Took care of both.